src/DssProxyActionsStETH.t.sol

```diff
@@ -0,0 +1,318 @@
+        spotter.poke("WSTETH");
+        (,, uint256 spot,,) = vat.ilks("WSTETH");
+        assertEq(spot, 50 * RAY * RAY / 1500000000 ether);
+
+        manager = new DssCdpManager(address(vat));
+        DSProxyFactory factory = new DSProxyFactory();
+        registry = new ProxyRegistry(address(factory));
+        dssProxyActions = address(new DssProxyActions());
+        dssProxyActionsEnd = address(new DssProxyActionsEnd());
+        dssProxyActionsStETH = address(new DssProxyActionsStETH(address(vat), address(manager)));
+        dssProxyActionsEndStETH = address(new DssProxyActionsEndStETH(address(vat), address(manager)));
+        proxy = DSProxy(registry.build());
+    }
+
+    function ink(bytes32 ilk, address urn) public view returns (uint256 inkV) {
+        (inkV,) = vat.urns(ilk, urn);
+    }
+
+    function art(bytes32 ilk, address urn) public view returns (uint256 artV) {
+        (,artV) = vat.urns(ilk, urn);
+    }
+
+    function testLockStETH() public {
+        uint256 cdp = this.open(address(manager), "WSTETH", address(proxy));
+        stETH.approve(address(proxy), 2 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)), 0);
+        uint256 prevBalance = wstETH.balanceOf(address(this));
+        this.lockStETH(address(wstETHJoin), cdp, 2 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)), wstETH.getWstETHByStETH(2 ether));
+        assertEq(stETH.balanceOf(address(this)), prevBalance - 2 ether);
+    }
+
+    function testSafeLockStETH() public {
+        uint256 cdp = this.open(address(manager), "WSTETH", address(proxy));
+        stETH.approve(address(proxy), 2 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)), 0);
+        uint256 prevBalance = wstETH.balanceOf(address(this));
+        this.safeLockStETH(address(wstETHJoin), cdp, 2 ether, address(proxy));
+        assertEq(ink("WSTETH", manager.urns(cdp)), wstETH.getWstETHByStETH(2 ether));
+        assertEq(stETH.balanceOf(address(this)), prevBalance - 2 ether);
+    }
+
+    function testLockStETHOtherCDPOwner() public {
+        uint256 cdp = this.open(address(manager), "WSTETH", address(proxy));
+        this.give(address(manager), cdp, address(123));
+        stETH.approve(address(proxy), 2 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)), 0);
+        uint256 prevBalance = wstETH.balanceOf(address(this));
+        this.lockStETH(address(wstETHJoin), cdp, 2 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)),wstETH.getWstETHByStETH(2 ether));
+        assertEq(stETH.balanceOf(address(this)), prevBalance - 2 ether);
+    }
+
+    function testFailSafeLockStETHOtherCDPOwner() public {
+        uint256 cdp = this.open(address(manager), "WSTETH", address(proxy));
+        this.give(address(manager), cdp, address(123));
+        stETH.approve(address(proxy), 2 ether);
+        this.safeLockStETH(address(wstETHJoin), cdp, 2 ether, address(321));
```

Comment:
I can't find an example for some reason, but I know I've reviewed tests were we use a `try/catch` to test the error message we get back for failing tests.  I prefer this since these tests are prone to false positives since any failure makes them look like a `success`.  If there is a fragility to the `try/catch` stuff since I know that is new/in development, then something Sam did in the D3M was leave a comment in the tests saying what the error message should be

Reply:
An example of what Chris is mentioning: https://github.com/makerdao/dss/blob/master/src/test/clip.t.sol#L535

Reply:
Mmm this doesn't really test the meeage, just makes sure the specific call is the one that's failing.

Reply:
Yeah Chris mentioned it as a second option to make sure the developer checks that the failing is happening on that specific revert that should happen in the last external call the test has.

Reply:
Done.